spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/ConfigServerHealthIndicator.java

```diff
@@ -53,6 +54,11 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 				HashMap<String, Object> detail = new HashMap<>();
 				detail.put("name", environment.getName());
 				detail.put("label", environment.getLabel());
+
+				if (environment.getCached()) {
+					builder.outOfService();
```

Comment:
Should we consider the config server out of service just because the repo couldnt be reached?  I mean technically it can still serve configuration data right?